include/xrpl/protocol/STObject.h

```diff
@@ -505,7 +507,26 @@ class STObject::Proxy
 // Constraint += and -= ValueProxy operators
 // to value types that support arithmetic operations
 template <typename U>
-concept IsArithmetic = std::is_arithmetic_v<U> || std::is_same_v<U, STAmount>;
+concept IsArithmeticNumber = std::is_arithmetic_v<U> ||
+    std::is_same_v<U, Number> || std::is_same_v<U, STAmount>;
```

Comment:
Shouldn't `IOUAmount`, `MPTAmount`, and `XRPAmount` be also included? They are addable.